core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -345,20 +345,21 @@ class LogSegment private[log] (val log: FileRecords,
    */
   @nonthreadsafe
   def truncateTo(offset: Long): Int = {
-    val mapping = translateOffset(offset)
-    if (mapping == null)
-      return 0
     offsetIndex.truncateTo(offset)
     timeIndex.truncateTo(offset)
     txnIndex.truncateTo(offset)
-    // after truncation, reset and allocate more space for the (new currently  active) index
+
+    // after truncation, reset and allocate more space for the (new currently active) index
     offsetIndex.resize(offsetIndex.maxIndexSize)
     timeIndex.resize(timeIndex.maxIndexSize)
-    val bytesTruncated = log.truncateTo(mapping.position)
-    if(log.sizeInBytes == 0) {
+
+    val mapping = translateOffset(offset)
```

Comment:
Would it be better to search the offset before truncating the index? Otherwise, when a log segment is truncated completely, translateOffset() will have to scan the full log segment in order to find the message because the offset index is empty.

Reply:
Ack